homeassistant/components/mill/coordinator.py

```diff
@@ -40,3 +52,105 @@ def __init__(
             update_method=mill_data_connection.fetch_heater_and_sensor_data,
             update_interval=update_interval,
         )
+
+
+class MillHistoricDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching Mill historic data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        *,
+        mill_data_connection: Mill,
+    ) -> None:
+        """Initialize global Mill data updater."""
+        self.mill_data_connection = mill_data_connection
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="MillHistoricDataUpdateCoordinator",
+        )
+
+    async def _async_update_data(self):
+        """Update historic data via API."""
+        now = dt_util.utcnow()
+        self.update_interval = (
+            timedelta(hours=1) + now.replace(minute=0, second=0) - now
+        )
+
+        for dev_id, heater in self.mill_data_connection.devices.items():
+            if not isinstance(heater, Heater):
+                continue
+            statistic_id = f"{DOMAIN}:energy_{slugify(dev_id)}"
+
+            last_stats = await get_instance(self.hass).async_add_executor_job(
```

Comment:
I think it would make sense to call `get_instance` once outside the loop instead of twice in each loop iteration.